sbt-plugin/src/main/scala/scala/scalajs/sbtplugin/ScalaJSPluginInternal.scala

```diff
@@ -632,6 +633,24 @@ object ScalaJSPluginInternal {
     }
   }
 
+  protected def javaSysProps2JSLib(javaOptions: Seq[String]): Option[VirtualJSFile] = {
+    val javaSysPropsPattern = "-D([^=]*)=(.*)".r
+    val formattedProps = javaOptions.collect {
+      case javaSysPropsPattern(propName, propValue) =>
+        "\"" + escapeJS(propName) + "\": \"" + escapeJS(propValue) + "\""
+    }
```

Comment:
For now, it can fail here, since we know that the Scala.js sbt plugin does not have other handlers. Using `javaOptions` to change different parts of the pipeline is just plain wrong (and will fail in JVM projects since java will complain).

If we need to understand other options, we probably need to introduce an intermediary task that parses these into an "JvmOptions" structure. This task can then fail, downstream tasks use the structured result.